lib/rules/no-assert-equal-boolean.js

```diff
@@ -0,0 +1,111 @@
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const EQUALITY_ASSERTIONS = ["equal", "deepEqual", "strictEqual"];
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "require use of boolean assertions",
+            category: "Best Practices"
+        },
+        fixable: "code",
+        messages: {
+            error: "Use assert.true(...) or assert.false(...) for boolean assertions."
+        },
+        schema: []
+    },
+
+    create: function (context) {
+        // Declare a test stack in case of nested test cases (not currently supported by QUnit).
+        const testStack = [];
+
+        function getCurrentAssertContextVariable() {
+            assert(testStack.length, "Test stack should not be empty");
+
+            return testStack[testStack.length - 1].assertVar;
+        }
+
+        // Check for something like `equal(...)` without assert parameter.
+        function isGlobalEqualityAssertion(calleeNode) {
+            return calleeNode &&
+                calleeNode.type === "Identifier" &&
+                EQUALITY_ASSERTIONS.includes(calleeNode.name);
+        }
+
+        // Check for something like `assert.equal(...)`.
+        function isAssertEquality(calleeNode) {
+            return calleeNode &&
+                calleeNode.type === "MemberExpression" &&
+                calleeNode.property.type === "Identifier" &&
+                EQUALITY_ASSERTIONS.includes(calleeNode.property.name) &&
+                calleeNode.object.type === "Identifier" &&
+                calleeNode.object.name === getCurrentAssertContextVariable();
+        }
+
+        // Check for something like `equal(...)` or `assert.equal(...)`.
+        function isEqualityAssertion(calleeNode) {
+            return isGlobalEqualityAssertion(calleeNode) ||
+                isAssertEquality(calleeNode);
+        }
+
+        // Finds the first boolean argument of a CallExpression if one exists.
+        function getBooleanArgument(node) {
+            return node.arguments.length >= 2 &&
+                [node.arguments[0], node.arguments[1]]
+                    .find(arg => arg.type === "Literal" && (arg.value === true || arg.value === false));
+        }
+
+        function reportError(node) {
+            context.report({
+                node: node,
+                messageId: "error",
+                fix(fixer) {
+                    const sourceCode = context.getSourceCode();
+
+                    // Decide which assertion function to use based on how many negations we have.
+                    const booleanArgument = getBooleanArgument(node);
+                    let countNegations = 0;
+                    if (!booleanArgument.value) {
+                        countNegations++;
+                    }
+                    const newAssertionFunctionName = countNegations % 2 === 0 ? "true" : "false";
```

Comment:
This could be simplified to `booleanArgument.value ? "true" : "false"`.

Reply:
Updated.